### PR TITLE
Use select instead of query for sqlalchemy

### PR DESCRIPTION
**Pull request by mcoughlin**
This PR begins to move the permissions infrastructure to use sqlalchemy's select instead of query.

### CODE REVIEW

**Comment by stefanv** [comment]
Hi Michael, thanks for this!

Since there are no baselayer tests, I'll have to rely on you testing this against SkyPortal (which I know you are doing).

Some questions:

1. I see sometimes select is used inside of session, sometimes not—how does that work?
2. Should we also move to the `with session` syntax?

Otherwise the changes look straightforward and we can merge this soon.  After that, I will enable black + pre-commit on this repo, so we don't have to worry about the formatting of all those long lines any more.

**Comment by mcoughlin** [comment]
@stefanv 
So in baselayer, sometimes the requirement is to return just the query (and therefore just the select) and the results of the query (and therefore the session). That's why that is happening.

We can do the new syntax, but I thought that might be a second round for SP at least.

File: app/models.py [hunk 1]
```diff
@@ -961,9 +959,11 @@ def query_records_accessible_by(
             )
 
         logic = getattr(cls, mode)
-        return logic.query_accessible_rows(
-            cls, user_or_token, columns=columns
-        ).options(options)
+        query = logic.query_accessible_rows(cls,
+                                            user_or_token, columns=columns)
+        for option in options:
+            query = query.options(option)
```
Comment (stefanv) [inline app/models.py:965]: Why the for loop instead of, say, `query.options(options)`?

**Review by stefanv — COMMENTED** [review]
Thanks, Michael! This looks good; happy to merge.  I left one quick question.

File: app/models.py (hunk 1 above)
Comment (mcoughlin) [reply app/models.py:965]: I found options wanted a single option and not a list when using select.

Comment (stefanv) [reply app/models.py:965]: But, but... https://docs.sqlalchemy.org/en/14/orm/query.html#sqlalchemy.orm.Query.options

Comment (mcoughlin) [reply app/models.py:965]: I see, maybe ** instead of just a list then?

Comment (stefanv) [reply app/models.py:965]: Isn't options a list?

Comment (mcoughlin) [reply app/models.py:965]: [16:35:36 basehandler] Error response returned by [/api/internal/recent_gcn_events]: [ExecutionOption Core or ORM object expected, got [<sqlalchemy.orm.strategy_options._UnboundLoad object at 0x1676b3d00>].]

when passing it a list.

Comment (stefanv) [reply app/models.py:965]: 🤷‍♂️

**Pull request merged**